src/main/java/seedu/weme/statistics/TagManager.java

```diff
@@ -0,0 +1,100 @@
+package seedu.weme.statistics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+import javafx.collections.ObservableList;
+
+import seedu.weme.model.ReadOnlyMemeBook;
+import seedu.weme.model.meme.Meme;
+import seedu.weme.model.tag.Tag;
+
+/**
+ * Tag manager in Statistics feature that reads tags from a {@code MemeBook}.
+ */
+public class TagManager {
+
+    private Set<Tag> tags;
+    private PriorityQueue<TagWithCount> tagsWithCount;
+
+    public TagManager() {
+        tags = new HashSet<>();
+        tagsWithCount = new PriorityQueue<>();
+    }
+
+    public Set<Tag> getTagsInSet() {
+        return tags;
+    }
+
+    /**
+     * Get {@code TagWithCount} in PriorityQueue from the pre-stored data.
+     */
+    public PriorityQueue<TagWithCount> getTagsWithCountInPriorityQ() {
+        return tagsWithCount;
+    }
+
+    /**
+     * Get {@code TagWithCount} in PriorityQueue from a {@code ReadOnlyMemeBook}.
+     */
+    public PriorityQueue<TagWithCount> getTagsInOrderOfCounts(ReadOnlyMemeBook memeBook) {
+        resetData();
+        parseMemeBookForTags(memeBook);
+        return tagsWithCount;
+    }
+
+    /**
+     * Returns {@code TagWithCount} in List.
+     */
+    public List<TagWithCount> getTagsWithCountList() {
+        List<TagWithCount> tagWithCountList = new ArrayList<>();
+        PriorityQueue<TagWithCount> temp = getTagsWithCountInPriorityQ();
+
+        while (!tagsWithCount.isEmpty()) {
+            tagWithCountList.add(temp.poll());
+        }
+
+        return tagWithCountList;
+    }
+
+    public void setTags(Set<Tag> tags) {
```

Comment:
What is this for?

Reply:
I can delete since it is not in use. It was written in case I need a set of Tags (for other stats features that needs it). (Maybe Zimu can use it somehow)

Reply:
i dont think you should be able to manually set tags, it should be parsed via memebook only since the parse method replaces it every time.

Reply:
Make sense. Will change now.